### ISSUE TITLE
Unable to reproduce the scores from the paper using the scripts you provide.

### ISSUE THREAD

**Issue by kanseaveg** [TIGER-AI-Lab/StructLM#8]
Hello, I tried to run the results of StructLM-7B using the script you provided, but the results are clearly inconsistent with the data in the paper.
![image](https://github.com/TIGER-AI-Lab/StructLM/assets/80022154/e248c4f0-2171-4890-b221-64bb39311d19)

The reproduced log and the reproduced summary are as follows.
[StructLM-7B-eval.log](https://github.com/TIGER-AI-Lab/StructLM/files/14571492/StructLM-7B-eval.log)
[summary.json](https://github.com/TIGER-AI-Lab/StructLM/files/14571493/summary.json)

Additionally, the generated prediction file is too large, so it won't be uploaded here.
![image](https://github.com/TIGER-AI-Lab/StructLM/assets/80022154/6c2e3a9a-d55a-4800-967d-525ee3bbc1b9)

I'm using the same environment as stated in the readme, with an A800 graphics card. **Why is there a significant discrepancy between my reproduced results and those reported in the paper?**

Additionally, in order to better identify the cause, I am providing the hash values of the weights and data that I downloaded from Huggingface.
![image](https://github.com/TIGER-AI-Lab/StructLM/assets/80022154/c7c8a5fb-3fb9-409a-b78e-f24e97c205e2)
![image](https://github.com/TIGER-AI-Lab/StructLM/assets/80022154/4e147e8c-94b6-4bdf-bcfd-919ea978e117)

**Comment by azhx** [comment]
Thanks for your comment. Let me reproduce this. I have a feeling this should be a small issue since most datasets experience no performance degradation.

**Comment by kanseaveg** [comment]
In addition, when I try structLM-13B, I find that loading will also appear error, please also help to take a look.

**Comment by azhx** [comment]
Hi, I'm able to load structLM-13B and begin the prediction process. Can you send me a log of your error?
<img width="614" alt="image" src="https://github.com/TIGER-AI-Lab/StructLM/assets/29048249/11fa592f-3596-4349-a449-ce740bfaf040">

I also just updated the repo to use a longer input token length, since the one specified in the eval script was too small

Also, I reproduced your StructLM-7B result. I'm investigating further. 

I evaluated StructLM-13B, and it closely reproduces the results in the paper.

**Comment by kanseaveg** [comment]
![image](https://github.com/TIGER-AI-Lab/StructLM/assets/80022154/512df56c-e085-4574-8cdc-762988a01536)

@azhx I'm really sorry that it cannot be used because of environmental problems, but now it can be run. 
If there is a huge discrepancy in the experimental results evaluated, I will still give feedback on this issue.
Thank you for your patient answer. By the way, have you found the reason for the inconsistent results of your Struct-7B?

**Comment by kanseaveg** [comment]
![image](https://github.com/TIGER-AI-Lab/StructLM/assets/80022154/78143290-57c7-45b1-95e9-d80cef2c667c)
For StructLM-13B, the scores evaluated by the breakpoint can all be correct, and there is no much difference. What I am curious about is that your average calculation is very different from the average in the paper. I suspect that there is a deviation between the calculation method and the calculation method of the average value in the code, remember to modify it. In addition, I would like to ask what problems exist in the effect of StructLM-7B again, and whether I need to provide the prediction file `predictions_predict.json` to you. Thank you for your patient answer and look forward to your reply again.

**Comment by azhx** [comment]
> What I am curious about is that your average calculation is very different from the average in the paper

Hi, the average calculation in the paper is just over the data reported in the main table. As you can see here, there are more than one metric reported for a many datasets, and those are less important. We omit them from the average. The main table averages are calculated the same way, so they are comparable. The metrics that we use in the main table are specified there.

> what problems exist in the effect of StructLM-7B 

At this point i suspect there is an issue with the checkpoint that was uploaded. The resources are tight on our cluster and I'm waiting for it to be freed up soon. I hope to solve this issue asap. Thanks for your continued interest, and I'll update this thread as soon as I make a change

**Comment by kanseaveg** [comment]
Thank you for your patient reply and look forward to the verification result of your 7B

**Comment by kanseaveg** [comment]
![image](https://github.com/TIGER-AI-Lab/StructLM/assets/80022154/fef28bc4-062b-4a21-a259-399d73037964)
Hi, the result of 34B is also out, which is completely consistent with that of the paper. 
I suspect that the gap in the Struct-7B may be due to the original `llamatrainer` training.
Anyway, thanks for your patience and help.
![image](https://github.com/TIGER-AI-Lab/StructLM/assets/80022154/1e1ce89b-e4ac-49ac-9f01-b4f954866388)
In addition, I found a curious phenomenon. For example, in the spider dataset, the results of 13B and 34B were surprisingly consistent.
Normally, 34B should be more emergent than 13B, but the effectiveness of instruct does not seem to be improved.
I think this is a question worth thinking about. @azhx

**Comment by azhx** [comment]
Hi, thanks for indicating the issue is resolved, but I will continue to update this thread when I am able to address the 7B issue. Sorry about the delay on that and thanks for your patience.

> For example, in the spider dataset, the results of 13B and 34B were surprisingly consistent.

Yes, my guess about this result is that data is more important than scaling model size for performance on this niche.

**Comment by azhx** [comment]
Hi @kanseaveg 
We have a new 7B model trained on Mistral that perfoms better than the original 7B codellama model and reproduces. please see https://huggingface.co/TIGER-Lab/StructLM-7B-Mistral/. 

Note that there is a different prompting format, so you should use the file at to run the evaluation
https://huggingface.co/datasets/TIGER-Lab/SKGInstruct/blob/main/skginstruct_test_file_mistral.json

let me know if you have any questions

**Comment by kanseaveg** [comment]
Thank you for your contribution to the community~ @azhx 
BTW. May I ask what is the difference between the instruct data of codellama and that of mistral?

**Comment by azhx** [comment]
The training data we used this time omits the slimorca, and it was trained with a different prompt format. You can check the model card for details
